gpytorch/kernels/inducing_point_kernel.py

```diff
@@ -60,8 +60,8 @@ def _get_covariance(self, x1, x2):
             covar = RootLazyTensor(k_ux1.matmul(self._inducing_inv_root))
 
             # Diagonal correction for predictive posterior
-            correction = (lazify(self.base_kernel(x1, x2)).diag() - covar.diag()).clamp(0, math.inf)
-            covar = PsdSumLazyTensor(DiagLazyTensor(correction), covar)
+            correction = (self.base_kernel(x1, x2, diag=True) - covar.diag()).clamp(0, math.inf)
+            covar = covar + DiagLazyTensor(correction)
```

Comment:
Why not PsdSumLazyTensor?

Reply:
This keeps things as an AddedDiagLazyTensor, but I have no strong opinions about it -- is there a good reason to prefer PsdSumLazyTensor over AddedDiagLazyTensor given that we'd have to do a root decomposition of `covar` anyways?

Reply:
`covar` is already a RootLazyTensor, so we can avoid an extra root decomposition

Reply:
Okay, I put PsdSumLazyTensor back in and simplified the for loop in inducing_inv_root further.